File: android/src/main/kotlin/com/superwall/superwallkit_flutter/bridges/SurveyBridge.kt
package com.superwall.superwallkit_flutter.bridges

import com.superwall.sdk.config.models.Survey
import com.superwall.sdk.config.models.SurveyOption
import com.superwall.sdk.config.models.SurveyShowCondition

fun Survey.toJson(): Map<String, Any?> {
    return mapOf(
        "id" to id,
        "assignmentKey" to assignmentKey,
        "title" to title,
        "message" to message,
        "options" to options.map { it.toJson() },
        "presentationCondition" to presentationCondition.toJson(),
        "presentationProbability" to presentationProbability,
        "includeOtherOption" to includeOtherOption,
        "includeCloseOption" to includeCloseOption
    )
}

fun SurveyOption.toJson(): Map<String, Any?> {
    return mapOf(
        "id" to id,
        "title" to title
    )
}

fun SurveyShowCondition.toJson(): String {
    return when (this) {
        SurveyShowCondition.ON_MANUAL_CLOSE -> "onManualClose"
        SurveyShowCondition.ON_PURCHASE -> "onPurchase"
    }
}

File: ios/Classes/Bridges/SurveyBridge.swift
import Flutter
import SuperwallKit

extension Survey {
  func toJson() -> [String: Any] {
    var dict = [String: Any]()
    dict["id"] = id
    dict["assignmentKey"] = assignmentKey
    dict["title"] = title
    dict["message"] = message
    dict["options"] = options.map { $0.toJson() }
    dict["presentationCondition"] = presentationCondition.toJson()
    dict["presentationProbability"] = presentationProbability
    dict["includeOtherOption"] = includeOtherOption
    dict["includeCloseOption"] = includeCloseOption
    return dict
  }
}

extension SurveyOption {
  func toJson() -> [String: Any] {
    var dict = [String: Any]()
    dict["id"] = id
    dict["title"] = title
    return dict
  }
}

extension SurveyShowCondition {
  func toJson() -> String {
    switch self {
      case .onManualClose:
        return "onManualClose"
      case .onPurchase:
        return "onPurchase"
    }
  }
}


